File: docs/sharepoint/codesnippet/VisualBasic/projecttasklist1/projecttasklisteventreceiver/projecttasklisteventreceiver.vb
Option Explicit On
Option Strict On

Imports System
Imports System.Security.Permissions
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Security
Imports Microsoft.SharePoint.Utilities
Imports Microsoft.SharePoint.Workflow

Public Class ProjectTaskListEventReceiver 
    Inherits SPItemEventReceiver

	''' <summary>
	''' An item was added.
    ''' </summary>
    '<snippet1>
    Public Overrides Sub ItemAdded(ByVal properties As SPItemEventProperties)
        MyBase.ItemAdded(properties)
        Dim web As SPWeb = properties.OpenWeb()
        properties.ListItem("Due Date") = "July 1, 2009"
        properties.ListItem("Description") = "This is a critical task."
        properties.ListItem.Update()
    End Sub
    '</snippet1>


End Class

File: docs/sharepoint/codesnippet/CSharp/projecttasklist/projecttasklisteventreceiver/projecttasklisteventreceiver.cs
using System;
using System.Security.Permissions;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Security;
using Microsoft.SharePoint.Utilities;
using Microsoft.SharePoint.Workflow;

namespace ProjectTaskList.ProjectTaskListEventReceiver
{
    /// <summary>
    /// List Item Events
    /// </summary>
    public class ProjectTaskListEventReceiver : SPItemEventReceiver
    {
       /// <summary>
       /// An item was added.
       /// </summary>
 
        //<snippet1>
        public override void ItemAdded(SPItemEventProperties properties)
       {
           base.ItemAdded(properties);
           SPWeb web = properties.OpenWeb();
           properties.ListItem["Due Date"] = "July 1, 2009";
           properties.ListItem["Description"] = "This is a critical task.";
           properties.ListItem.Update(); 
       }  
        //</snippet1>

    }
}
